model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java

```diff
@@ -62,6 +62,12 @@ public void update(T entity) {
 
     public void delete(T entity) {
         if (entity != null) {
+            // making sure the entity in question,
+            // is really part of this transaction
+            if (! entityManager.contains(entity)) {
+                entity = entityManager.merge(entity);
```

Comment:
Please be sure we test this also with installations associated to
categories (and categories that are shared by other installations as well)

On Wed, Apr 15, 2015 at 1:42 PM, Erik Jan de Wit <notifications@github.com>
wrote:

> In
> model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/529#discussion_r28412266>
> :
>
> > @@ -62,6 +62,12 @@ public void update(T entity) {
> >
> >      public void delete(T entity) {
> >          if (entity != null) {
> > +            // making sure the entity in question,
> > +            // is really part of this transaction
> > +            if (! entityManager.contains(entity)) {
> > +                entity = entityManager.merge(entity);
>
> This will load the entity from the database, better would be:
>
> entityManager.getReference(getType(), entity.getId());
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/529/files#r28412266>
> .
>